crates/sui-node/src/lib.rs

```diff
@@ -1115,6 +1119,44 @@ impl SuiNode {
         assert_eq!(next_epoch, new_epoch_store.epoch());
         new_epoch_store
     }
+
+    fn check_system_consistency(&self) {
+        if !self.config.enable_expensive_safety_checks && cfg!(not(debug_assertions)) {
+            return;
+        }
+        let total_storage_rebate: u64 = self
+            .state
+            .db()
+            .iter_live_object_set()
+            .map(|o| {
+                if let LiveObject::Normal(object) = o {
+                    object.storage_rebate
```

Comment:
Should we go ahead and go `object.get_total_sui()` here? That we can also check that the total is 10B and we'll cover everything.

Reply:
Yes we can. Although that will really depend on everything to be fixed first